### PR TITLE
Fixing ancor links for non-latin scripts

### PR DESCRIPTION
**Pull request by Andygol**
Anchor link fixing

### CODE REVIEW

File: _layouts/doc.html [hunk 1]
```diff
@@ -61,7 +61,7 @@
   var addId = function(list){
     for (i = 0; i < list.length; i++) {
       element = list[i];
-      sectionCopy = element.innerHTML.toLowerCase().replace(/ /g, '-').replace(/([^0-9a-z-])/g,''); //substitute space with dash
```
Comment (althio) [inline _layouts/doc.html]: `.replace(/([^0-9a-z-])/g,'')`

What was is used for? @lxbarth @jeffhaack @jueyang
Will it break something if we remove this replacement/removal function for non-latin characters?

@Andygol 
What is the effect if you before/after for non-latin scripts?

**Comment by Andygol** [comment]
@althio 
in current state (with `.replace(/([^0-9a-z-])/g,'')`) we can't get a right anchor for non-latin script. 

With this issue I faced making translation into Ukrainian, Russian

Instead getting `http://learnosm.org/ru/beginner/start-osm/#посетите-сайт-openstreetmap`  we have `http://learnosm.org/ru/beginner/start-osm/#--openstreetmap` that not always pointed user to a right location. Or other example instead of `/#перемещение-по-карте` we get `/#--`

I made change in my local repo and http://andygol.github.io/learnosm/ru/beginner/start-osm/#посетите-сайт-openstreetmap point to the right place :smile:

**Comment by althio** [comment]
@lxbarth @jeffhaack @jueyang
Will it break something if we remove this replacement/removal function for non-latin characters?

**Comment by althio** [comment]
I still cannot make my mind on this.
Is this similar to http://sourceforge.net/p/docutils/feature-requests/42 ?
What is the best practice around?

**Comment by Andygol** [comment]
@althio 
Yes, this is the stuff you are mentioned above. I tried to fix it, but as I see for now not very well. Here we need to use approach  that is allow make links not only for latin scripts

**Comment by Nick-Tallguy** [comment]
Closing this now. Would appear some kind of fix was implemented as Russian and other non Latin scripts are present.

**Comment by althio** [comment]
We could close as stale.

**Please note however it is not fixed.**

This only affects anchor links (`link to this section`) on pages with non Latin script.
Links are not human readable and sometimes they collide and are just wrong.

A few examples:

current | should be
---|---
[ru/beginner/start-osm/#--openstreetmap](http://learnosm.org/ru/beginner/start-osm/#--openstreetmap) | [ru/beginner/start-osm/#посетите-сайт-openstreetmap](http://andygol.github.io/learnosm/ru/beginner/start-osm/#посетите-сайт-openstreetmap)
[ru/beginner/start-osm/#--](http://learnosm.org/ru/beginner/start-osm/#--) | [ru/beginner/start-osm/#просмотр-карты-различными-стилями](http://andygol.github.io/learnosm/ru/beginner/start-osm/#просмотр-карты-различными-стилями)
[ru/beginner/start-osm/#----](http://learnosm.org/ru/beginner/start-osm/#----) | [ru/beginner/start-osm/#добавление-ваших--первых-точек](http://andygol.github.io/learnosm/ru/beginner/start-osm/#добавление-ваших--первых-точек)
[ru/beginner/start-osm/#](http://learnosm.org/ru/beginner/start-osm/#) | [ru/beginner/start-osm/#итоги](http://andygol.github.io/learnosm/ru/beginner/start-osm/#итоги)
[ja/beginner/start-osm/#openstreetmap](http://learnosm.org/ja/beginner/start-osm/#openstreetmap) | [ja/beginner/start-osm/#openstreetmapウェブサイトの表示](http://andygol.github.io/learnosm/jp/beginner/start-osm/#openstreetmapウェブサイトの表示)
[ja/beginner/start-osm/#openstreetmap](http://learnosm.org/ja/beginner/start-osm/#openstreetmap) | [ja/beginner/start-osm/#openstreetmapのアカウント登録](http://andygol.github.io/learnosm/jp/beginner/start-osm/#openstreetmapのアカウント登録)

**Comment by Nick-Tallguy** [comment]
OK - leaving closed as scale - seeing the comment that Andy made that the fix was not right.

**Comment by Nick-Tallguy** [comment]
stale, not scale!

**Comment by althio** [comment]
The proposed fix is quite easy.
Could we test on staging site and see if anything breaks?

**Comment by Nick-Tallguy** [comment]
I'll try to test it on the staging site tonight

**Comment by Nick-Tallguy** [comment]
I've pulled in the pull request to the staging site, but had to make a couple of changes;
  1. Quite a few conflicts sorted - many things have moved in 4 years,  
  2. The liquid syntax has altered and I've had to remove a couple of { } and .  to make my system run locally without throwing error messages - I've had to do this for other files in the past, so this was no surprise.  
 3. I've put a file in from the main site so we have something with links in it to look at = http://nick-tallguy.github.io/ru/josm/start-josm/  (Sorry, different to the one you referred to, but it still has sub=headings).  I can add start-osm if you need it.

I've had a quick look, but there seems to be a problem - you can no longer click on a sub-heading to get the hyperlink for it - nothing displays in the address bar at all now.  

Please make sure I haven't done anything when I fixed the conflicts etc.  I'll leave on the staging site for a day or two to allow checking and perhaps amending.

**Comment by althio** [comment]
> Please make sure I haven't done anything when I fixed the conflicts etc. I'll leave on the staging site for a day or two to allow checking and perhaps amending.

Looking right now, not sure I understand why it does not reproduce the links as before.
Any thoughts @Andygol ?

**Comment by Andygol** [comment]
@althio I'll take a look on this. A lot of things happened during this time 😏 …

**Comment by althio** [comment]
@Nick-Tallguy @Andygol 
It works for me on the staging site.
http://nick-tallguy.github.io/ru/josm/start-josm/
Let me know what you see.

**Comment by Andygol** [comment]
@althio 
👍 ✨ and for me too. I've just checked and it works as expected.
This link <http://nick-tallguy.github.io/ru/josm/start-josm/#изучение-основ-рисования-в-josm> is working 🎉

**Comment by Andygol** [comment]
It would be great to fix this case (<http://nick-tallguy.github.io/ru/josm/start-josm/#точки,-линии-и-фигуры-(полигоны)>) too, when we have <kbd>,</kbd> (coma) in the title
@althio @Nick-Tallguy

**Comment by Nick-Tallguy** [comment]
Hi @Andygol & @althio - I've just had a play and the fix in the pull request is pretty good! Thank you and well done. 

**Now we need to get it into Learnosm.**

Although I could copy the code in, it may be better if;  
1. We close this pull request, but then 
2.  @Andygol does a new pull request and we straight away merge this in (when I say new, I mean with the fork up to date so we don't have to mess around sorting the conflicts), 
3. In addition, please could you do it for doc-rtl as well, so that it works for Persian and any other **right to left** languages.

This way Andy gets the credit for the code. 

But if you haven't got time, then thanks for doing this anyway and I'll copy it across in a few days.

Regards and thanks again

Nick

**Comment by Nick-Tallguy** [comment]
forgot to add - I've got the ru & uk modules which were part of the pull request - would you like me to add them in, or ....

**Comment by Andygol** [comment]
@Nick-Tallguy 
I hope it is good just to close this PR in favor of new one with no any other action

**Comment by Andygol** [comment]
New PR is here https://github.com/hotosm/learnosm/pull/530

**Pull request closed without merging**